core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -757,7 +767,7 @@ private[spark] class AppStatusStore(
         executorLogs,
         AppStatusUtils.schedulerDelay(taskDataOld),
         AppStatusUtils.gettingResultTime(taskDataOld))
-    }.toSeq
+    }.toList
```

Comment:
Out of curiosity, why this change?

Reply:
Let me double check this

Reply:
This change is to make `HistoryServerSuite` passed,  `toSeq` causes `LevelDB/RocksDBIterator` close in advance, while 'toList' does not. 

[de2862e](https://github.com/apache/spark/pull/36237/commits/de2862ec2fd1caec1c74b092510982b287795b58) try to fixed the issue in a different way